stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py

```diff
@@ -584,24 +589,23 @@ async def execute_search(
             raise NotFoundError(f"Collections '{collection_ids}' do not exist")
 
         hits = es_response["hits"]["hits"]
-        items = (hit["_source"] for hit in hits)
+        items = (hit["_source"] for hit in hits[:limit])
```

Comment:
Looking good. Could you explain why you use [:limit] here? Are the hits returned automatically limited to the limit? Or should this be [:limit-1]?

Reply:
The query uses: `size_limit = min(limit + 1, max_result_window)` so we try to query +1 item

Say a user queries with `limit = 10`

We will query for `10+1 = 11`

So the number of items we want to return is `10` or `[:limit]`